src/apps/service/apps/de/listings.clj

```diff
@@ -40,17 +40,20 @@
    (augment-listing-params params short-username (perms-client/load-app-permissions short-username))))
 
 (defn- augment-search-params
-  [search_term params short-username app-ids]
-  (let [category-attrs (set (workspace-metadata-category-attrs))]
-    (if (or (empty? app-ids) (empty? category-attrs))
-      params
-      (assoc params :pre-matched-app-ids
-                    (metadata-client/filter-targets-by-ontology-search short-username
-                                                                       (categorization/get-active-hierarchy-version)
-                                                                       category-attrs
-                                                                       search_term
-                                                                       ["app"]
-                                                                       app-ids)))))
+  [search_term {:keys [app-ids public-app-ids] :as params} short-username admin?]
+  (let [category-attrs (set (workspace-metadata-category-attrs))
+        app-ids        (if admin? public-app-ids app-ids)]
+    (-> params
+        (assoc :app-ids             app-ids
+               :pre-matched-app-ids (when-not (or (empty? app-ids) (empty? category-attrs))
```

Comment:
This is an extremely trivial suggestion, but I'd probably write something like:

```
(when (every? seq [app-ids category-attrs]) ...)
```

or

```
(when-not (some empty? [app-ids category-attrs]) ...)
```

It's not functionally different, though; it's just a personal preference. Feel free to ignore this suggestion, of course. :)